### PR TITLE
Dummy Pull Request to Provoke CI

### PR DESCRIPTION
**Pull request by geoffthemedio**
not for merging...

### CODE REVIEW

**Comment by geoffthemedio** [comment]
@agrrr3 comments on this?

**Comment by agrrr3** [comment]
...I moved the comment to the review..

**Review by agrrr3 — COMMENTED** [review]
i am bit confused about the PR title and the relationship to refactor_test branch

i try to remember why i wrote the stuff like i did.

const valueref like you did makes a lot more sense than a const pointer to a valueref.

having toplevelcontent as a member is as ugly as always, but the implementation to set with the mutable getter is fine. Guess it works only with (the most relevant) int and double currently because there are no other explicit specialisations of NamedValueRefManager::GetMutableValueRef? Or do those get generated?

made some review comments to the code, else looks good to me

File: universe/NamedValueRefManager.cpp [hunk 1]
```diff
@@ -47,35 +47,54 @@ NamedValueRefManager::NamedValueRefManager() {
...
+template const ValueRef::ValueRef<UniverseObjectType>* NamedValueRefManager::GetValueRef(const std::string&, bool) const;
+template const ValueRef::ValueRef<StarType>* NamedValueRefManager::GetValueRef(const std::string&, bool) const;
 
 
-ValueRef::ValueRefBase* const NamedValueRefManager::GetValueRefBase(const std::string& name) const {
-    auto* drefp = const_cast<NamedValueRefManager*>(this)->GetValueRef<double>(name);
-    //if (auto* drefp = const_cast<NamedValueRefManager*>(this)->GetValueRef<double>(name)) // TODO C++17
-    if (drefp)
+const ValueRef::ValueRefBase* NamedValueRefManager::GetValueRefBase(const std::string& name) const {
+    if (auto* drefp = GetValueRef<double>(name))
```
[the remark this answers is not in the document]
Comment (geoffthemedio) [reply universe/NamedValueRefManager.cpp:95]: Without `::` to select the free function, it's unambiguously calling the member function.

If there is no template specialization for a type, the default template implementation in the header is instantiated for that type.

The default and specialized implementations all call `CheckPendingNamedValueRefs()`

File: universe/NamedValueRefManager.cpp [hunk 2]
```diff
@@ -47,35 +47,54 @@ NamedValueRefManager::NamedValueRefManager() {
...
 }
 
-template ValueRef::ValueRef<PlanetEnvironment>* const NamedValueRefManager::GetValueRef(const std::string&, const bool);
-template ValueRef::ValueRef<PlanetType>* const NamedValueRefManager::GetValueRef(const std::string&, const bool);
-template ValueRef::ValueRef<Visibility>* const NamedValueRefManager::GetValueRef(const std::string&, const bool);
-template ValueRef::ValueRef<PlanetSize>* const NamedValueRefManager::GetValueRef(const std::string&, const bool);
-template ValueRef::ValueRef<UniverseObjectType>* const NamedValueRefManager::GetValueRef(const std::string&, const bool);
-template ValueRef::ValueRef<StarType>* const NamedValueRefManager::GetValueRef(const std::string&, const bool);
+template <>
+ValueRef::ValueRef<int>* NamedValueRefManager::GetMutableValueRef(const std::string& name,
```
[the remark this answers is not in the document]
Comment (geoffthemedio) [reply universe/NamedValueRefManager.cpp:68]: `GetMutableValueRef<T>` is in the header https://github.com/freeorion/freeorion/pull/3343/files#diff-358bb8e2e021b8c31f3e650586443813e79bb65c974755ba78e72192112136afR242

`GetMutableValueRefBase` sounds reasonable but probably in a separate change.

**Pull request closed without merging**